internal/services/systemcentervirtualmachinemanager/system_center_virtual_machine_manager_cloud_resource.go

```diff
@@ -0,0 +1,209 @@
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+type SystemCenterVirtualMachineManagerCloudModel struct {
+	Name                                                   string            `tfschema:"name"`
+	Location                                               string            `tfschema:"location"`
+	ResourceGroupName                                      string            `tfschema:"resource_group_name"`
+	CustomLocationId                                       string            `tfschema:"custom_location_id"`
+	SystemCenterVirtualMachineManagerServerInventoryItemId string            `tfschema:"system_center_virtual_machine_manager_server_inventory_item_id"`
+	Tags                                                   map[string]string `tfschema:"tags"`
+}
+
+var _ sdk.Resource = SystemCenterVirtualMachineManagerCloudResource{}
+var _ sdk.ResourceWithUpdate = SystemCenterVirtualMachineManagerCloudResource{}
+
+type SystemCenterVirtualMachineManagerCloudResource struct{}
+
+func (r SystemCenterVirtualMachineManagerCloudResource) ModelObject() interface{} {
+	return &SystemCenterVirtualMachineManagerCloudModel{}
+}
+
+func (r SystemCenterVirtualMachineManagerCloudResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return clouds.ValidateCloudID
+}
+
+func (r SystemCenterVirtualMachineManagerCloudResource) ResourceType() string {
+	return "azurerm_system_center_virtual_machine_manager_cloud"
+}
+
+func (r SystemCenterVirtualMachineManagerCloudResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validate.SystemCenterVirtualMachineManagerCloudName,
+		},
+
+		"location": commonschema.Location(),
+
+		"resource_group_name": commonschema.ResourceGroupName(),
+
+		"custom_location_id": commonschema.ResourceIDReferenceRequiredForceNew(&customlocations.CustomLocationId{}),
+
+		"system_center_virtual_machine_manager_server_inventory_item_id": commonschema.ResourceIDReferenceRequiredForceNew(&inventoryitems.InventoryItemId{}),
+
+		"tags": commonschema.Tags(),
+	}
+}
+
+func (r SystemCenterVirtualMachineManagerCloudResource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{}
+}
+
+func (r SystemCenterVirtualMachineManagerCloudResource) Create() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			subscriptionId := metadata.Client.Account.SubscriptionId
+			client := metadata.Client.SystemCenterVirtualMachineManager.Clouds
+
+			var model SystemCenterVirtualMachineManagerCloudModel
+			if err := metadata.Decode(&model); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			scvmmServerInventoryItemId, err := inventoryitems.ParseInventoryItemIDInsensitively(model.SystemCenterVirtualMachineManagerServerInventoryItemId)
```

Comment:
insensitively parsing should be used for API response only. Configured properties should be sensitively parsed.

Reply:
Updated